src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretService.java

```diff
@@ -116,33 +121,30 @@ public boolean encryptAndSaveSecret(OnChainObjectType onChainObjectType,
                                         String secretOwner,
                                         String secretKey,
                                         String secretValue) {
-        if (isSecretPresent(onChainObjectType, onChainObjectAddress, secretOwnerRole, secretOwner, secretKey)) {
+        try {
             final TeeTaskComputeSecret secret = TeeTaskComputeSecret
                     .builder()
                     .onChainObjectType(onChainObjectType)
                     .onChainObjectAddress(onChainObjectAddress)
                     .secretOwnerRole(secretOwnerRole)
                     .fixedSecretOwner(secretOwner)
                     .key(secretKey)
+                    .value(encryptionService.encrypt(secretValue))
                     .build();
-            log.info("Tee task compute secret already exists, can't update it." +
-                    " [secret:{}]", secret);
+            log.info("Adding new tee task compute secret [secret:{}]", secret);
+            int result = jdbcTemplate.update("INSERT INTO \"tee_task_compute_secret\" "
+                            + "(\"on_chain_object_type\", \"on_chain_object_address\", \"secret_owner_role\", \"fixed_secret_owner\", \"key\", \"value\") VALUES "
+                            + "(?, ?, ?, ?, ?, ?)",
+                    onChainObjectType.ordinal(), onChainObjectAddress.toLowerCase(), secretOwnerRole.ordinal(), secretOwner.toLowerCase(), secretKey, encryptionService.encrypt(secretValue));
+            if (result != 1) {
+                return false;
+            }
```

Comment:
I would like to have a log in such a case. debug level if `result == 0`, warn level if `result > 1`. This log should include the header and the number of results.

Reply:
After our discussions, let's apply the following: drop the `result > 1` case as it can never happen, and log the case `result == 0` as error.